agent/command/s3_put.go

```diff
@@ -140,18 +102,12 @@ type s3put struct {
 	// workDir sets the working directory relative to which s3put should look for files to upload.
 	// workDir will be empty if an absolute path is provided to the file.
 	workDir          string
-	skipMissing      bool
 	preservePath     bool
 	skipExistingBool bool
```

Comment:
nit: while we're refactoring, can we just call this `skipExisting`? It's just the bool version of `SkipExisting`.

Reply:
Yes!

> I don't mind it if we want to refactor to consolidate common functionality, but in the future, can we split it into two PRs if the refactor is significant? Most of the changes are to refactor existing logic, which we could do more incrementally, whereas the actual change (rename the role ARN field) is rather tiny.

That's super fair, I originally wasn't going to clean up the commands but it was getting more and more frustrating keeping them in sync. Next time feel free to push back on it harder (you can quote me!)

Reply:
I'm going to put off the refactor and reopen with just the change for the ticket